lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FocusEvent, KeyboardEvent, MouseEvent, Ref, useCallback, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import { always, any, clamp, dec, inc, pipe, prop } from 'ramda';
+
+import './NxCombobox.scss';
+
+import forwardRef from '../../util/genericForwardRef';
+import { Props, propTypes } from './types';
+import NxTextInput from '../NxTextInput/NxTextInput';
+import NxDropdownMenu from '../NxDropdownMenu/NxDropdownMenu';
+import NxLoadWrapper from '../NxLoadWrapper/NxLoadWrapper';
+import { useUniqueId } from '../../util/idUtil';
+import useMutationObserver from '@rooks/use-mutation-observer';
+import DataItem from '../../util/DataItem';
+export { Props } from './types';
+
+function NxComboboxRender<T extends string | number = string>(
+  props: Props<T>,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+        className: classNameProp,
+        loading,
+        error,
+        matches,
+        onSelect,
+        searchText,
+        onSearchTextChange,
+        onSearch,
+        long,
+        short,
+        disabled,
+        emptyMessage,
+        autoComplete,
+        inputProps,
+        id,
+        'aria-required': ariaRequired,
+        'aria-describedby': ariaDescribedBy,
+        ...attrs
+      } = props,
+
+      isEmpty = !matches.length,
+
+      dropdownRef = useRef<HTMLDivElement>(null),
+      inputRef = useRef<HTMLDivElement>(null),
+      elFocusedOnMostRecentRender = useRef<Element | null>(null),
+
+      [focusableBtnIndex, setFocusableBtnIndex] = useState<number | null>(null),
+      [elToFocusId, setElToFocusId] = useState(''),
```

Comment:
I suspect something isn't right about `elToFocusId`, but first could you explain what its purpose is?  Two things confuse me about it:
1. It seems to roughly implement the same thing that `focusableBtnIndex` already implements: tracking which list item is focusable
2. It is used as the value for `aria-activedescendant` but is set as an id of any element. `aria-activedescendant` is supposed to be set to the DOM id of the "focused" child